com.unity.render-pipelines.high-definition/Editor/Material/ShaderGraph/HDSubTarget.cs

```diff
@@ -94,19 +98,25 @@ public override void Setup(ref TargetSetupContext context)
             if (!context.HasCustomEditorForRenderPipeline(typeof(HDRenderPipelineAsset)))
                 context.AddCustomEditorForRenderPipeline(customInspector, typeof(HDRenderPipelineAsset));
 
-            if (migrationSteps.Migrate(this))
-                OnBeforeSerialize();
+            if (m_MigrateFromOldSG)
+            {
+                systemData.version = ShaderGraphVersion.Initial;
```

Comment:
The upgrade system was broken here, and there was a weird hack to workaround it, but it should be fixed correctly now